ci/test/check-merge-with-target.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+#
+# check-merge-with-target.sh — check if merge with target branch succeeds and checks still pass.
+
+set -euo pipefail
+
+. misc/shlib/shlib.bash
+
+git fetch "$BUILDKITE_REPO"
```

Comment:
This might fetch the whole history and waste some CPU/time/bandwidth. Could we fetch less of the history and still merge fine?

On a quick look I couldn't find a way to find the base commit of a PR using buildkite. With that we could `git fetch --shallow-exclude=$BASE_COMMIT`.

Github might have this information behind a REST API, but I'm not sure if it's worth the effort.

Reply:
Apparently the `git fetch` only takes 1 s, so I assume we have most of the history already, then let's just leave it as is.

Reply:
✅ I have changed it to just fetch the relevant branch from the remote repo.